org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java

```diff
@@ -2056,6 +2116,30 @@ protected String buildDataSQL(int start, int end) {
 		return dataSql;
 	}
 
+    /**
+     * Join tables required for the ORDER BY clause
+     * @param orderClause
+     * @return SQL JOIN
+     */
+    private String getSQLJoin(String sqlMain) {
+	if(joinTables.size() <= 0)
+		return "";
+	StringBuilder builder = new StringBuilder();
+	for(String tableName : joinTables) {
+		boolean doJoin = true;
+		for(TableInfo tableInfo : tableInfos) {
+			if(tableName.equalsIgnoreCase(tableInfo.getTableName()))
+				doJoin = false;
+		}
+		if(doJoin && !sqlMain.contains(" JOIN " + tableName)) {
+			builder.append(" JOIN ").append(tableName).append(" ON (")
```

Comment:
inner join doesn't always work, the column could be optional and contains NULL value.

Reply:
Ok, so I went with LEFT JOIN instead